### ISSUE TITLE
Modularize org-roam features

### ISSUE THREAD

**Issue by progfolio** [jethrokuan/org-roam#357]
org-roam has a lot of potential for growth. It would make sense to break some functionality out into its own feature/file. For example, the org-roam-graph functionality seems like a good starting point. It is pretty well contained and could be made more consistent/generic so that its possible to support graphing software other than Graphviz in the future. Users who do not need the graphing functionality could skip the load time for that feature as well. Org does something similar with `org-modules'.

**Comment by progfolio** [comment]
Looking over the source I feel like the database, graphing and capture logic could all be made into their own features.

**Comment by progfolio** [comment]
I realize this is a very large commit, but the first step is getting each feature into its own file and normalizing the function/variable names:

https://github.com/progfolio/org-roam/commit/5bbcf5e0326ec7eabab676416731ccc3ea737cdd

Happy to work on this more if you think it's a viable direction.

**Comment by jethrokuan** [comment]
Most of the features that can be moved into their own files are already separated by in sections using `;;;`.

I think moving things into their own files is good for maintainability. But I'd like a defalias to  the original function names, at least for the interactives, so people's keybindings don't break.

**Comment by progfolio** [comment]
> Most of the features that can be moved into their own files are already separated by in sections using ;;;.

I followed roughly that pattern.

> I think moving things into their own files is good for maintainability. But I'd like a defalias to the original function names, at least for the interactives, so people's keybindings don't break.

I've added the command aliases and rebased on top of https://github.com/jethrokuan/org-roam/commit/df29da1b6db56d44f0f1232815612e8084dc177d

Here: https://github.com/progfolio/org-roam/tree/refactor-modularize-features

I'm running makem.sh locally and I've addressed all relevant errors:

```
Starting new Ispell process /usr/bin/aspell with default dictionary... \ 
Starting new Ispell process /usr/bin/aspell with default dictionary...done
company-org-roam.el:60: Error checking word UNIX using aspell with default dictionary
company-org-roam.el:118: Error checking word BACKEND using aspell with default dictionary
company-org-roam.el:132: Error checking word BACKEND using aspell with default dictionary
company-org-roam.el:139: Error checking word BACKEND using aspell with default dictionary
org-roam-capture.el:190: Error checking word PREPENDS using aspell with default dictionary
org-roam-capture.el:246: Error checking word ORG-ROAM using aspell with default dictionary
org-roam-completion.el:51: Error checking word ORG-ROAM using aspell with default dictionary
org-roam-db.el:50: Error checking word SQLITE using aspell with default dictionary
org-roam-db.el:68: Error checking word SQLITE using aspell with default dictionary
org-roam-db.el:81: Error checking word ENTRYPOINT using aspell with default dictionary
org-roam-db.el:81: Error checking word SQLITE using aspell with default dictionary
org-roam-db.el:113: Error checking word EMACSQL using aspell with default dictionary
org-roam-graph.el:50: Error checking word GRAPHER using aspell with default dictionary
org-roam-graph.el:55: Error checking word GRAPHER using aspell with default dictionary
org-roam-graph.el:57: Error checking word RANKDIR using aspell with default dictionary
org-roam-graph.el:67: Error checking word MATCHER using aspell with default dictionary
org-roam-graph.el:71: Error checking word MATCHER using aspell with default dictionary
org-roam-graph.el:88: Error checking word SQL using aspell with default dictionary
org-roam-graph.el:114: Error checking word GRAPHER using aspell with default dictionary
org-roam-protocol.el:29: There should be two spaces after a period
org-roam-protocol.el:49: Error checking word URL using aspell with default dictionary
org-roam-protocol.el:53: Error checking word JAVASCRIPT using aspell with default dictionary
org-roam-protocol.el:53: Error checking word HREF using aspell with default dictionary
org-roam-protocol.el:54: Error checking word ENCODEURICOMPONENT using aspell with default dictionary
org-roam-protocol.el:55: Error checking word ENCODEURICOMPONENT using aspell with default dictionary
org-roam-protocol.el:56: Error checking word ENCODEURICOMPONENT using aspell with default dictionary
org-roam-protocol.el:76: Error checking word EMACSCLIENT using aspell with default dictionary
org-roam.el:82: Error checking word FORMATTER using aspell with default dictionary
org-roam.el:82: Error checking word FORMATTER using aspell with default dictionary
org-roam.el:99: Error checking word GPG using aspell with default dictionary
org-roam.el:129: Error checking word OX-HUGO using aspell with default dictionary
org-roam.el:129: Error checking word HTTPS using aspell with default dictionary
org-roam.el:130: Error checking word GITHUB using aspell with default dictionary
org-roam.el:130: Error checking word KAUSHALMODI using aspell with default dictionary
org-roam.el:130: Error checking word OX-HUGO using aspell with default dictionary
org-roam.el:130: Error checking word BC using aspell with default dictionary
org-roam.el:130: Error checking word BCA using aspell with default dictionary
org-roam.el:130: Error checking word OX-HUGO using aspell with default dictionary
org-roam.el:130: Error checking word EL using aspell with default dictionary
org-roam.el:227: Error checking word EMACSQL using aspell with default dictionary
org-roam.el:564: Error checking word BACKLINKS using aspell with default dictionary
org-roam.el:572: Error checking word BACKLINKS using aspell with default dictionary
org-roam.el:740: Error checking word BACKLINKS using aspell with default dictionary
ERROR (2020-03-27 16:49:10): Linting checkdoc failed.
org-roam-completion.el:96: Indentation mismatch
org-roam-completion.el:97: Indentation mismatch
org-roam-completion.el:98: Indentation mismatch
org-roam-completion.el:99: Indentation mismatch
org-roam-completion.el:100: Indentation mismatch
ERROR (2020-03-27 16:49:12): Linting indentation failed.
Entering directory '/home/n/.emacs.d/straight/repos/org-roam/'
org-roam-db.el:73:0: error: Aliases should start with the package's prefix "org-roam-db".
org-roam-db.el:191:0: error: Aliases should start with the package's prefix "org-roam-db".
org-roam-db.el:353:0: error: Aliases should start with the package's prefix "org-roam-db".
org-roam-graph.el:173:0: error: Aliases should start with the package's prefix "org-roam-graph".
org-roam-macs.el:36:0: error: "org-roam--with-temp-buffer" doesn't start with package's prefix "org-roam-macs".
ERROR (2020-03-27 16:49:13): Linting package failed.
LOG (2020-03-27 16:49:17): Finished with 3 errors.
```
The complaint about the macro not starting with the org-roam-macs prefix is not something I'd worry about. Org-mode has a similar convention of using an `org-macs.el` file for macros which does not follow the prefix convention either.

The alias prefix complaints will exist until support is dropped for the original command aliases. The new commands start with each feature's proper prefix.

The only thing I did not address is the documentation/CHANGELOG.

**Comment by jethrokuan** [comment]
This should be closed with #363
